### PR TITLE
Add Mousetrap mission (scu33ea from Counterstrike)

### PR DESCRIPTION
**Pull request by JovialFeline**
This adds Mousetrap, where Soviets hunt General Stavros in a bunker full of Chronospheres.

----

### Changes and Quirks

This is not an _exhaustive_ list, but it should be close. Despite the short length, I found the mission complex (with branching paths, even! Dang). I made a flowchart that details much of the logic as I dug through it, [found here if that sounds useful](https://i.imgur.com/IHhpyEd.png). It helped keep my head straight.

Infantry weapon range is reduced 5c0 -> 4c0, partly to avoid targeting Stavros through thick walls (when he's at the control room's western console in particular). The mission focuses on infantry v. infantry, so I don't expect this to feel too impactful. If it is, I'm open to fixing that.

**Original:** The escape plan for Stavros does little. Everything seems set up for it but rather than chronoshifting to his goal, he will attempt (and fail) walking there. If this was fixed in Retaliation or the remaster, I wouldn't know.
  - **Newish:** If possible, Stavros will go to the escape vehicle. The conditions for escape:
    - The starting room's console was not used.
    - Self-destruct was started.
    - Stavros was not pushed back into the control room.

**Original:** The starting console is disabled after its use, or immediately when Soviets move past the first crate room.
  - **New:** If unused, Stavros will release gas near the console. This is done at the same time he releases gas near the power plants. The idea was to add some clarity to why it can't be used (unless protected by the curtain crate, anyway).

**Other:**
- A soldier is added near the APC. If the general escapes, this soldier will enter as well. Their death by gas will prompt the APC to leave. They're absent in the original but `die2` seems to reference a soldier dying from this gas.
- Chronospheres will move a dummy unit around to show the building effects when appropriate. The palette shift is removed to mirror original behavior, and so players don't go cross-eyed from all the shifting.
- After its use, the damaged Chronosphere "misfires" before blowing up as usual.
- Troops emerge from Chronospheres 1s after the sound starts, instead of at the same time. Stavros still enters and emerges from Chronospheres over a 4s span (when compared to default 5/7 speed on Normal).
- To match `soviet-07`, only two-legged soldiers can use consoles.
- The `endguys` team spawns from a Chronosphere in the northeast corner, rather than the neighboring sphere used by Stavros.

### Hard stuff

Normal's more of a straight conversion. I don't know of any written guidelines for what Hard ought to look like, but I've tried going for a New Game+ feel. There are some additions and twists for a (hopefully) interesting replay after Normal. The mission's (infinite) reinforcements do a lot to ensure victory anyway.
<details><summary>Details:</summary>

- Some defenses are built near the start. They'll need clearing or at least distracting.
- A spy will attempt to kill all the (non-shifted) engineers and retreat for the Soviet entrance. YAmismo's commando in Twist of Fate (https://github.com/OpenRA/OpenRA/pull/20743) gave me this idea.
- The north hall's unused Chronosphere now spawns enemies until destroyed. This is the unused `atk4` team on a shortened timer (~90s -> 60s), and from a different Chronosphere. A short camera reveal happens upon each shift.
- Stavros has a second, smaller defense team near the last stand. Both teams will converge once any team member is attacked.
- Stavros' himself is more threatening, though a poor match for more than a couple of soldiers.
</details>

----

### Original mission trivia:

- While Stavros waits on Soviets to enter the east hallway, a Chronosphere can be shot to get his attention. He'll attempt to move there and defend the building. The mission will not end once he dies.
- Another soft lock: activate self-destruct as Stavros is moving to or around the control room. He'll continue to hide there.
- It's possible to shoot and kill Stavros immediately after a shift from the control room. I left this unchanged; if a player pulls this off, they likely know the map well. Good for them!
- Three rifles (`atk3`) seem intended to chronoshift into the southern crate room once Stavros makes it to a console. This does not function.
- `atk4` is an unused group of infantry spawned from a Chronosphere in the northeast corner. Its trigger is on a repeating timer. The flag for this trigger is never set, but it is cleared once Stavros' last stand is reinforced.
- Entering the east hallway triggers gas on that side. The control room's eastmost console should also release this gas, but it's never touched.
- The escape APC's `chin` and `chin2` teams imply this was a helicopter at some point.
- The starting room's Chronosphere used to have a self-destruct console. (Trigger `dst1`)

### Personal nonsense

I'm quite new to Git(Hub) and far from a seasoned programmer. While I've tried to follow the norms I see in other maps and keep the script organized, I expect this to need changes. Thanks for your time and eyeballs.

I'll give shouts to Dr. Über for volunteering as a test monkey, penev92 for answering questions, and whoever was responsible for `Actor.CallFunc`, which saved me some headache.

### CODE REVIEW

**Comment by PunkPun** [comment]
Needs a rebase and difficulty translations added. See changelog #20585 for how.

**Comment by JovialFeline** [comment]
- Rebased with updated translation rules.
- Fixed errors and warnings in `rules.yaml` prompted by Stavros and the tweaked Chronospheres.
- Fixed `GroupHuntFromRally` not starting the hunt if too many units were involved.
- `CheckRequiredUnits` no longer has a (_tiny_) chance to queue a loss right after the Soviets' victory.
- If a Chronosphere is not passed to `ChronoEffect`, the dummy is no longer shifted. Only the notification sound will be played.

**Comment by JovialFeline** [comment]
- Removed an Easy option that does nothing. This was added along with the difficulty translations. Oops.
- It was possible to capture the damaged sphere, repair it with a second engineer, then sell the building before its intended death to cause a crash. The crash is gone, and the sale will no longer spawn infantry.
- Stavros should now react appropriately to invincible dogs that go straight to the eastern hallway.

**Comment by PunkPun** [comment]
I managed to bug out the mission. If at the very start I move the iron curtained dogs towards the stravos chronospehere and trigger a premature stravos retreat, he will then bug out and will never leave his chamber. (Not even after selfdestruct is initialised).

* On normal difficulty I couldn't trigger a stravos retreat via apc.
* On hard. All reinforcements are useless because of the pillboxes. I'm not sure what should be done to fix that.
* On hard the only way I managed to win is with the dog cheese, where I clear everything with iron curtained dogs and wait for stravos to retreat after the sabotage

**Comment by JovialFeline** [comment]
- Patched what were hopefully the last two opportunities to break Stavros' pathing.
- Traded Hard's pair of pillboxes for one. If the pillbox is left intact, it will later be destroyed by `SelfDestructSequence`.

I couldn't repeat the Normal escape problem on my end.

**Comment by PunkPun** [comment]
The chronosphere at the spawn doesn't seem to blow up, same with chrono at the apc.

Otherwise LGTM. Didn't check lua.

**Comment by JovialFeline** [comment]
> The chronosphere at the spawn doesn't seem to blow up, same with chrono at the apc.

For whatever reason, that was original behavior. I'm not opposed to adding more explosions, of course.

File: mods/ra/maps/mousetrap/rules.yaml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+World:
+	LuaScript:
+		Scripts: campaign.lua, utils.lua, Mousetrap.lua
```
Comment (Mailaender) [inline mods/ra/maps/mousetrap/rules.yaml]:
```suggestion
		Scripts: campaign.lua, utils.lua, mousetrap.lua
```

Linux is case sensitive and will crash.

Comment (JovialFeline) [reply mods/ra/maps/mousetrap/rules.yaml]: Thank you for the correction. Fixed.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f3a98c39174e1be977db3c47bf49cd4a94bd0494...a1770687a2a3c5fa012fcb076e2866dfcded23e3)

**Pull request merged**